CarpetX/src/io_openpmd.cxx

```diff
@@ -1195,23 +1206,59 @@ void carpetx_openpmd_t::InputOpenPMD(const cGH *const cctkGH,
                 constexpr std::uint64_t ipoison =
                     0xfff8000000000000ULL + 0xdeadbeef;
 #endif
-                CCTK_REAL poison;
-                std::memcpy(&poison, &ipoison, sizeof poison);
+                const size_t typesize = size_t(CCTK_VarTypeSize(cgroup.vartype));
+                std::vector<char> poison(typesize);
+                // for int: deadbeef for little endian machines
+                for (size_t i = 0; i < typesize; i += sizeof poison)
+                  std::memcpy(&poison[i], &ipoison, std::min(poison.size(), typesize - i));
```

Comment:
I would define three different poison values here (for int, real, and complex). Using just some of the `real` bits for `int` might give values that aren't very distinctive. I found that large positive values make for good integer poison. Negative values don't work that well because they are more likely to be clipped off.

Reply:
done in 49de23b